test/tests/functional/pbs_resource_usage_log.py

```diff
@@ -0,0 +1,350 @@
+
+        # No R record ; Only E record for job3 which was deleted
+        # when in 'R' state
+        self.server.accounting_match(
+            msg='.*R;' + jid3, id=jid3, existence=False,
+            regexp=True, max_attempts=10)
+        self.server.accounting_match(
+            msg='.*E;' + jid3 +
+            '.*Exit_status=271.*resources_used.*run_count=1', id=jid3,
+            regexp=True)
+
+    def test_acclog_mom_down(self):
+        """
+        Check accounting logs when node is down and MoM is restarted
+        """
+        a = {ATTR_nodefailrq: 15}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        a = {'resources_available.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        # Submit a job
+        a = {'Resource_List.select': '1:ncpus=1:mem=20gb'}
+        j = Job(TEST_USER, a)
+        j.create_eatcpu_job()
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid)
+
+        # Submit a job array
+        ja = Job(TEST_USER, attrs={
+            ATTR_J: '1-2',
+            'Resource_List.select': 'ncpus=1:mem=20gb'}
+        )
+        ja.create_eatcpu_job()
+        jid_a = self.server.submit(ja)
+
+        subjid1 = j.create_subjob_id(jid_a, 1)
+        subjid2 = j.create_subjob_id(jid_a, 2)
+
+        self.server.expect(JOB, {'job_state': 'R'}, subjid1)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid2)
+
+        self.assertTrue(self.server.isUp())
+        self.assertTrue(self.mom.isUp())
+
+        # kill -9 mom
+        self.mom.signal('-KILL')
+
+        # Verify that nodes are reported to be down.
+        self.server.expect(NODE, {ATTR_NODE_state: 'down'},
+                           id=self.mom.shortname)
+
+        time.sleep(15)
+
+        self.server.expect(JOB, {ATTR_state: 'Q'}, jid)
+        self.server.expect(JOB, {'job_state': 'Q'}, subjid1)
+        self.server.expect(JOB, {'job_state': 'Q'}, subjid2)
+
+        self.server.tracejob_match(
+            msg='Job requeued, execution node .* down', id=jid,
+            interval=5, regexp=True)
+        self.server.tracejob_match(
+            msg='Job requeued, execution node .* down', id=subjid1,
+            interval=5, regexp=True)
+        self.server.tracejob_match(
+            msg='Job requeued, execution node .* down', id=subjid2,
+            interval=5, regexp=True)
+
+        # now restart mom
+        self.mom.start()
+        self.assertTrue(self.mom.isUp())
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid)
+
+        self.server.delete(jid, wait=True)
+        self.server.delete(jid_a, wait=True)
+
+        # job1 has 2 R records and a E record
+        matches = self.server.accounting_match(
+            msg='.*R;' + jid + '.*Exit_status=0.*resources_used.*run_count=1',
+            id=jid, regexp=True, allmatch=True)
```

Comment:
Why are there 2 'R' records?  You only requeued jid once.  Plus, you're checking for run_count=1.  If the job was run multiple times, run_count would be bumped.

Reply:
2 R records get generated by design per job. Do you think it is a bug?

Reply:
It sounds like one to me.  You should get one 'R" record per time you requeue a job.  If run_count is 1, then you've only run the job once.

Reply:
I checked this and it is by design that 2 R records (with run_count=1) are generated. When mom_logs show “Job requeued, execution node … down”, immediately a ‘R’ record is written, and when job is force requeued/rerun another ‘R’ record is written, run_count staying at 1.

Reply:
OK, thanks for checking on it.